### PR TITLE
feat!: stellar-config crate

### PR DESCRIPTION
**Pull request by willemneal**
This adds an new crate `stellar-config` which contains all logic around configuration. Currently this just a refactor but in the future it could use the [`config`](https://crates.io/crates/config) crate to more formally describe the config.

BREAKING CHANGE:
This also removes the deprecated `config` subcommand since we had split it up already into `keys` and `network`.

### CODE REVIEW

**Review by elizabethengelman — COMMENTED** [review]
This is looking great!

File: cmd/crates/stellar-config/src/locator.rs [hunk 1]
```diff
@@ -351,8 +349,17 @@ fn global_config_path() -> Result<PathBuf, Error> {
...
 }
 
-impl Pwd for Args {
-    fn set_pwd(&mut self, pwd: &Path) {
-        self.config_dir = Some(pwd.to_path_buf());
+/// # Errors
+/// May not find a config dir
+pub fn find_config_dir(mut pwd: std::path::PathBuf) -> std::io::Result<std::path::PathBuf> {
+    let soroban_dir = |p: &std::path::Path| p.join(".soroban");
+    while !soroban_dir(&pwd).exists() {
```
[the remark this answers is not in the document]
Comment (willemneal) [reply cmd/crates/stellar-config/src/locator.rs:356]: You only want to pop if `<pwd>/.soroban` didn't exist as it mutates `pwd` to be its parent. It is a weird way and perhaps should look at how `cargo` finds `.cargo`.

File: cmd/crates/stellar-config/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+rust-version.workspace = true
+
+[dependencies]
+stellar-xdr = { workspace = true, features = ["cli"] }
+soroban-env-host = { workspace = true }
+soroban-spec = { workspace = true }
+stellar-strkey = { workspace = true }
+soroban-sdk = { workspace = true }
+soroban-rpc = { workspace = true }
+# config_inner =  {version = "0.14.0", package = "config", features = ["toml"]}
```
[the remark this answers is not in the document]
Comment (willemneal) [reply cmd/crates/stellar-config/Cargo.toml:20]: Yeah left that to remind me to use that crate in the future but an issue is better than a comment.

File: cmd/crates/stellar-config/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+[package]
+name = "stellar-config"
```
Comment (leighmcculloch) [inline cmd/crates/stellar-config/Cargo.toml:2]: I think we should scope this to the CLI. Stellar config is an overloaded term.
```suggestion
name = "stellar-cli-config"
```

**Review by leighmcculloch — COMMENTED** [review]
What's the goal driving breaking out the config part of the CLI into its own crate?

My initial assumption was that the goal was to move config logic into a library, for use in other settings, but it looks like the PR is moving CLI types such as clap args into the lib, which means I think this would be tightly coupled with the CLI.

**Comment by janewang** [comment]
config work waiting for the alias PR to go in.

**Comment by janewang** [comment]
blocked by the alias work

**Comment by willemneal** [comment]
closed in favor of #1479

**Pull request closed without merging**